Failed to run test locally for `riscv64gc-unknown-linux-gnu`

```
> TARGET=riscv64gc-unknown-linux-gnu ./ci/run.sh

error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:~/.cargo/bin:~/.vscode-server/cli/servers/Stable-38c31bc77e0dd6ae88a4e9cc93428cc27a56ba40/server/bin/remote-cli:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl" VSLANG="1033" "cc" "/tmp/rustcfJC1Gz/symbols.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.3nbrh6hn0p9o7c11f9a6tjcl3.rcgu.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.4flob5f7nsd9iwega7x8j3m9c.rcgu.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.4p3g9p3u4nal5hwfc4mi60bna.rcgu.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.52ht7ty5obp9za7o4sl7xn8s1.rcgu.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.7gpbzmrqxftpxgd3fdxwxpudx.rcgu.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.chbzfxuvp59q3xtr3dwwjdcr2.rcgu.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.ekgu1rj9tb7ttvirkw44xm71j.rcgu.o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.4pgcf8rb7aeo7xhokrcp6m5c7.rcgu.o" "-Wl,--as-needed" "-Wl,-Bstatic" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libtest-2dab0adc00210d36.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libgetopts-920cc1fdb73d5165.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libunicode_width-a20fa998bbd26821.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/librustc_std_workspace_std-fe050ef324dff49c.rlib" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/libstdarch_test-c7de4ec54b01d90c.rlib" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/librustc_demangle-ddf454688e80e1ed.rlib" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/liblazy_static-3bfc8979c28779de.rlib" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/libstd_detect-17e3871cf23db9a3.rlib" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/liblibc-8d7673db76df669b.rlib" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/libcfg_if-6bfb957a80631848.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libstd-b9d7741b35af4b59.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libpanic_unwind-af2cd11a92c2bb83.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libobject-5d539404328a7adb.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libmemchr-8a0a66ccaf46156c.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libaddr2line-0e910ba4ded90b7f.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libgimli-7395b8959c2b308f.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/librustc_demangle-5e6b80cc028f93d8.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libstd_detect-4fef4bffebf27a61.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libhashbrown-ecb5a25d23a1e266.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/librustc_std_workspace_alloc-0f74e4d13eb264b8.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libminiz_oxide-8b56891185ca50a0.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libadler-db6fa615b8f3336e.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libunwind-3f7b97794ac65cc4.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libcfg_if-0bfe290517692780.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/liblibc-d556e9c135451033.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/liballoc-1a86613516847afa.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/librustc_std_workspace_core-b2e7689c9e13dce4.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libcore-dfa9ebf0523e3763.rlib" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib/libcompiler_builtins-11cdff4381b94d6e.rlib" "-Wl,-Bdynamic" "-lc" "-lm" "-lrt" "-lpthread" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/riscv64gc-unknown-linux-gnu/lib" "-o" "~/dev/rust-dev/stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-nodefaultlibs"
  = note: /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: Relocations in generic ELF (EM: 243)
          /usr/bin/ld: /stdarch/target/riscv64gc-unknown-linux-gnu/release/deps/core_arch-c316a73a926311f3.1tadp9s53loap9vuit9qozcqb.rcgu.o: error adding symbols: file in wrong format
          collect2: error: ld returned 1 exit status
```

```
> TARGET=riscv64gc-unknown-linux-gnu cargo +nightly test --release -p core_arch
error: unknown riscv target feature: avx512vl
     --> crates/core_arch/src/x86/avx512bw.rs:20917:5
      |
20917 |     #[simd_test(enable = "avx512bw,avx512vl")]
      |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      |
      = note: this error originates in the macro `is_riscv_feature_detected` which comes from the expansion of the attribute macro `simd_test` (in Nightly builds, run with -Z macro-backtrace for more info)

(... numerous similar errors)
```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -70,7 +70,7 @@ to improve the documentation of `stdarch`!
 
 # Alternative Testing Instructions
 
-It is generally recommended that you use `ci/run.sh` to run the tests.
+It is generally recommended that you use `ci/run-docker.sh` to run the tests.
 However this might not work for you, e.g. if you are on Windows.
 
 In that case you can fall back to running `cargo +nightly test` and `cargo +nightly test --release -p core_arch` for testing the code generation.
@@ -79,7 +79,7 @@ In particular you need to set the `TARGET` environment variable as you would for
 In addition you need to set `RUSTCFLAGS` (need the `C`) to indicate target features, e.g. `RUSTCFLAGS="-C -target-features=+avx2"`.
 You can also set `-C -target-cpu=native` if you're "just" developing against your current CPU.
 
-Be warned that when you use these alternative instructions, [things may go less smoothly than they would with `ci/run.sh`][ci-run-good], e.g. instruction generation tests may fail because the disassembler named them differently, e.g. it may generate `vaesenc` instead of `aesenc` instructions despite them behaving the same.
+Be warned that when you use these alternative instructions, [things may go less smoothly than they would with `ci/run-docker.sh`][ci-run-good], e.g. instruction generation tests may fail because the disassembler named them differently, e.g. it may generate `vaesenc` instead of `aesenc` instructions despite them behaving the same.
 Also these instructions execute less tests than would normally be done, so don't be surprised that when you eventually pull-request some errors may show up for tests not covered here.
 
 

